### PR TITLE
Should fix #369 and #370

### LINKED ISSUE #369: The Promise returned by jpm/lib/post finishes too early
`jpm/lib/post` returns a Promise that finishes when the XPI is created, rather than when the XPI is posted.

### PR DESCRIPTION
**Pull request by FayneAldan**
Fixes #369 and #370

### CODE REVIEW

File: lib/post.js [hunk 1]
```diff
@@ -4,43 +4,38 @@
...
-      });
-      client.on('error', function (e) {
-        // if net errors need to be propagated.
-      });
-      client.on('close', function (had_error) {
-        console.log(had_error ? "Posting XPI to %s failed" : "Posted XPI to %s", postURL);
-        fs.unlink(xpiPath);
-        console.log("Removed XPI from " + xpiPath);
+  return xpi(manifest, extend(options, { skipUpdateRDF: true })).then(function (xpiPath) {
+		return when.promise(function(done) {
```
Comment (kumar303) [inline lib/post.js]: this should only be indented two spaces past the enclosing statement

File: lib/post.js [hunk 2]
```diff
@@ -4,43 +4,38 @@
...
+          client.end(buffer);
+        });
+        client.on('data', function (data) {
+          // Extension Auto-Installer does not close the connection, so we do it after 500ms
+          setTimeout(function() {
+            client.destroy();
+          }, 500);
+        });
+        client.on('error', function (e) {
+          // if net errors need to be propagated.
```
Comment (kumar303) [inline lib/post.js]: while you're in here, you might as well change the signature to `when.promise(function(resolve, reject) {...` so you can call `reject(e)` right here. That will show a stack trace in the console and will allow the caller to handle the promise rejection.

**Comment by kumar303** [comment]
I guess `post` doesn't have any tests so you're off the hook ;) This needs rebasing on master for the test suite fix though.

File: lib/post.js (hunk 1 above)
Comment (FayneAldan) [inline lib/post.js]: My bad. I normally use tabs when programming, instead of spaces.

File: lib/post.js [hunk 3]
```diff
@@ -4,43 +4,39 @@
 "use strict";
 
 var extend = require("lodash").extend;
-var execute = require("./task");
-var http = require("http");
 var xpi = require("./xpi");
-var cp = require('child_process');
```
Comment (kumar303) [inline lib/post.js]: thanks for cleaning these up. We should really start running a linter.

**Comment by kumar303** [comment]
Looks ok to me, thanks

**Pull request merged**